tests/test_result.py

```diff
@@ -375,3 +375,55 @@ async def to_err_async(i: int) -> Result[int, int]:
 # Lambda versions of the same functions, just for test/type coverage
 sq_lambda: Callable[[int], Result[int, int]] = lambda i: Ok(i * i)
 to_err_lambda: Callable[[int], Result[int, int]] = lambda i: Err(i)
+
+
+# Tests for do notation
```

Comment:
Let's just move these tests to a new file rather than do this, `test_result_do.py` perhaps?